### PR TITLE
Fixed file content duplication issue

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/edx-sga/issues/229

#### What's this PR do?
- Fixes an issue appears when student upload a big file, then other use upload another file.
- but other file is replaced by first file

#### How should this be manually tested?
- upload file big files from 2 students
- download them on staff view
- delete state of those students
- try again above 3 steps

@pdpinch

### CODE REVIEW

File: edx_sga/constants.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """Constants"""
 
-BLOCK_SIZE = 8 * 1024
+BLOCK_SIZE = 2**10 * 8  # 8kb
```
Comment (pdpinch) [inline edx_sga/constants.py]: Are you just trying to make this easier to read?

Comment (amir-qayyum-khan) [reply edx_sga/constants.py]: yes, this was old value in system,

**Comment by pdpinch** [comment]
@gsidebo or @noisecapella can you review this?

File: edx_sga/sga.py [hunk 2]
```diff
@@ -249,6 +249,7 @@ def upload_assignment(self, request, suffix=''):
         user = self.get_real_user()
         require(user)
         upload = request.params['assignment']
+        sha1 = get_sha1(upload.file)
```
Comment (noisecapella) [inline edx_sga/sga.py]: What's the significance of this change?

File: edx_sga/sga.py [hunk 3]
```diff
@@ -269,8 +268,11 @@ def upload_assignment(self, request, suffix=''):
         student_item_dict = self.get_student_item_dict()
         submissions_api.create_submission(student_item_dict, answer)
         path = self.file_storage_path(sha1, upload.file.name)
-        if not default_storage.exists(path):
-            default_storage.save(path, File(upload.file))
+        log.info("Saving file: %s at path: %s for user: %s", upload.file.name, path, user.username)
+        if default_storage.exists(path):
+            # save latest submission
+            default_storage.delete(path)
```
Comment (noisecapella) [inline edx_sga/sga.py]: Shouldn't the file not be there at all? When would it already exist?

File: edx_sga/sga.py (hunk 2 above)
Comment (amir-qayyum-khan) [reply edx_sga/sga.py]: some how file.tell() is effecting sha1 calculating, after the sha1 algo sends same hash. which is replacing files

File: edx_sga/sga.py (hunk 3 above)
Comment (amir-qayyum-khan) [reply edx_sga/sga.py]: yes, this is for 2nd time upload, user can upload as mush as he want before finalizing assignment.

**Comment by amir-qayyum-khan** [comment]
@noisecapella please check my comments

**Comment by noisecapella** [comment]
Just a reminder to also change `SpooledTemporaryFile` to `TemporaryFile`

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella

**Comment by amir-qayyum-khan** [comment]
@noisecapella i am not able to write on temp file. it seems i am doing something wrong

**Pull request merged**